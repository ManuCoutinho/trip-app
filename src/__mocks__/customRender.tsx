import { render } from '@testing-library/react'
import { RenderOptions } from '@testing-library/react/types'
import { ThemeContextProvider } from 'contexts/ThemeContext'
import { ReactElement, FC, ReactNode } from 'react'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
