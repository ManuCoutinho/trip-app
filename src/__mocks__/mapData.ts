import { MapProps } from 'components/Map/types'

export default {
  places: [
    {
      id: 'first-place',
      name: 'Rio de Janeiro',
      slug: 'rio-de-janeiro',
      location: {
        latitude: -22.925453,
        longitude: -43.277763
      }
    },
    {
      id: 'second-place',
      name: 'Lagos',
      slug: 'lagos',
      location: {
        latitude: 6.460649,
        longitude: 3.333309
      }
    }
  ]
} as MapProps
