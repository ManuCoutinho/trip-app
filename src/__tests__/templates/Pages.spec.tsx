import { render } from '__mocks__/customRender'
import PagesTemplate from 'templates/Pages'

const templatePropsMock = {
  heading: 'Heading Template test',
  body: 'Body template test mock'
}
describe('<PagesTemplate />', () => {
  it('should render the Pages template correctly', () => {
    const { getByRole, getByText } = render(
      <PagesTemplate {...templatePropsMock} />
    )
    const backLink = getByRole('link', { name: /link to home/i })
    const body = getByText(templatePropsMock.body)
    const heading = getByText(templatePropsMock.heading)
    expect(backLink).toBeInTheDocument()
    expect(body).toBeInTheDocument()
    expect(heading).toBeInTheDocument()
  })
  it('should match to snapshot', () => {
    const { container } = render(<PagesTemplate {...templatePropsMock} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
