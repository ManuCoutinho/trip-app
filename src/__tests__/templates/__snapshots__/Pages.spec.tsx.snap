// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PagesTemplate /> should match to snapshot 1`] = `
<section
  class="sc-gswNZR bRTHbR"
>
  <div
    class="sc-bcXHqe eCYpHH"
    data-testid="back-link"
  >
    <a
      href="/"
      title="link to Home"
    >
      <svg
        aria-label="Home"
        fill="currentColor"
        height="32"
        viewBox="0 0 256 256"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <line
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="200"
          x2="56"
          y1="56"
          y2="200"
        />
        <line
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="200"
          x2="56"
          y1="200"
          y2="56"
        />
      </svg>
    </a>
  </div>
  <h1
    class="sc-dkrFOg cqzPDp"
  >
    Heading Template test
  </h1>
  <div
    class="sc-hLBbgP ikdcep"
  >
    <div>
      Body template test mock
    </div>
  </div>
</section>
`;
